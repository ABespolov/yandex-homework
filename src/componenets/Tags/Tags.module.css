.tags{

}

.tag {
  padding: 3px 10px 6px 10px;
  background: rgba(51, 51, 51, 0.1);
  border-radius: 16px;
}

.tag:not(:first-child){
  margin-left: 12px;
}
