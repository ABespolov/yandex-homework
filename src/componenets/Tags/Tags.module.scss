.tag {
  display: inline-block;
  max-width: 236px;
  padding: 3px 12px 6px 12px;
  overflow: hidden;
  font-size: 14px;
  white-space: nowrap;
  text-overflow: ellipsis;
  background: rgba(51, 51, 51, 0.1);
  border-radius: 16px;
}

.tag:not(:first-child) {
  margin-left: 12px;
}

@media (max-width: 1200px) {
  .tag {
    max-width: 140px;
    margin-top: 14px;
  }
}
