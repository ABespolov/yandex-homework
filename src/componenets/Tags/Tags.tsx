import React, {useEffect, useRef} from "react";
import styles from "./Tags.module.css";

interface Tags {
  tags: string[];
  tagsList: string[];
}

export const Tags: React.FC<Tags> = ({tags, tagsList}) => {
  // @ts-ignore
  const currentTags = tags && tags.map((item) => <span className={styles.tag}>{tagsList[item].tag}</span>);
  return (
    <div className={styles.tags}>
      {currentTags}
    </div>
  );
};
