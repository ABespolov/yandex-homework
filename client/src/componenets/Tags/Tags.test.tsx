import {configure, shallow} from "enzyme";
import React from "react";
import Adapter from "enzyme-adapter-react-16";
import {Tags} from "./Tags";

configure({ adapter: new Adapter() });

describe("SingleLetterControls:", () => {
    it("Tags render: ", () => {
        const tags = [{tag: 'test1'}, {tag: 'test2'}];
        const component = shallow(<Tags tags={tags} />);
        expect(component).toMatchSnapshot();
    });
});
