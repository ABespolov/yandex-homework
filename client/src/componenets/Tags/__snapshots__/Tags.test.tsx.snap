// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleLetterControls: Tags render:  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tags
    tags={
      Array [
        Object {
          "tag": "test1",
        },
        Object {
          "tag": "test2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="tag"
        >
          test1
        </span>,
        <span
          className="tag"
        >
          test2
        </span>,
      ],
      "className": "tags",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "test1",
          "className": "tag",
        },
        "ref": null,
        "rendered": "test1",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "test2",
          "className": "tag",
        },
        "ref": null,
        "rendered": "test2",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="tag"
          >
            test1
          </span>,
          <span
            className="tag"
          >
            test2
          </span>,
        ],
        "className": "tags",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "test1",
            "className": "tag",
          },
          "ref": null,
          "rendered": "test1",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "test2",
            "className": "tag",
          },
          "ref": null,
          "rendered": "test2",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
